src/test/java/lambda/part1/exercise/Exercise3.java

```diff
@@ -0,0 +1,74 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise3 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + expression-lambda
+        Arrays.sort(persons, (person1, person2)->Integer.compare(person1.getAge(), person2.getAge()));
+
+        assertArrayEquals(new Person[]{
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort + statement-lambda
+        Arrays.sort(persons, (person1, person2)->{final String firstSirname = person1.getLastName();
+            final String secondSirname = person2.getLastName();
+            final String firstName = person1.getFirstName();
+            final String secondName = person2.getFirstName();
+            return firstSirname.compareTo(secondSirname)>0 ||
+                    firstSirname.compareTo(secondSirname)==0 && firstName.compareTo(secondName)>0? 1:
```

Comment:
It's better to make ```firstSirname.compareTo(secondSirname)``` operation **once** and save calculated value to variable (1 calculation instead 3). 

Anyway, it's hard to read this 4 lines.
You could use something like:
```Java
int lastNamesComparing = personLeft.getLastName().compareTo(personRight.getLastName();

return lastNamesComparing != 0 ? lastNamesComparing : personLeft.getFirstName().compareTo(personRight.getFirstName());
```
or like that:
```Java
String leftFullName = personLeft.getLastName() + personLeft.getFirstName();
String rightFullName = personRight.getLastName() + personRight.getFirstName();

return leftFullName.compareTo(rightFullName);
```